signals.c

```diff
@@ -29,36 +31,38 @@ static void SIGSEGV_handler(int sig, siginfo_t *si, void *unused)
 	log_FTL_version(true);
 
 	logg("Received signal: %s", strsignal(sig));
-	logg("     at address: %lu", (unsigned long) si->si_addr);
+	logg("     at address: %p", si->si_addr);
 	switch (si->si_code)
 	{
 		case SEGV_MAPERR: logg("     with code: SEGV_MAPERR (Address not mapped to object)"); break;
 		case SEGV_ACCERR: logg("     with code: SEGV_ACCERR (Invalid permissions for mapped object)"); break;
 #if defined(SEGV_BNDERR)
 		case SEGV_BNDERR: logg("     with code: SEGV_BNDERR (Failed address bound checks)"); break;
 #endif
-		default: logg("     with code: Unknown (%i), ",si->si_code); break;
+		default: logg("     with code: Unknown (%i)", si->si_code); break;
 	}
 
+// Check GLIBC availability as MUSL does not support live backtrace generation
+#if defined(__GLIBC__)
 	// Try to obtain backtrace. This may not always be helpful, but it is better than nothing
 	void *buffer[255];
 	const int calls = backtrace(buffer, sizeof(buffer)/sizeof(void *));
+	logg("Backtrace:");
+
 	char ** bcktrace = backtrace_symbols(buffer, calls);
 	if(bcktrace == NULL)
+		logg("Unable to obtain backtrace symbols!");
+
+	for(int j = 0; j < calls; j++)
 	{
-		logg("Unable to obtain backtrace (%i)!",calls);
-	}
-	else
-	{
-		logg("Backtrace:");
-		int j;
-		for (j = 0; j < calls; j++)
-		{
-			logg("B[%04i]: %s",j,bcktrace[j]);
-		}
+		logg("B[%04i]: %p, %s", j, buffer[j],
+		     bcktrace != NULL ? bcktrace[j] : "---");
 	}
-	free(bcktrace);
-
+	if(bcktrace != NULL)
+		free(bcktrace);
+#else
+	logg("!!! INFO: pihole-FTL has not been compiled for glibc !!!");
```

Comment:
This should have something about not being compiled with backtrace support, because on its own this message does not say what would have happened if it had been compiled with glibc. Maybe something more like:
```
!!! INFO: pihole-FTL has not been compiled with backtrace support, not generating one !!!
```